gcloud/test_connection.py

```diff
@@ -66,6 +67,26 @@ def test_user_agent_format(self):
         conn = self._makeOne()
         self.assertEqual(conn.USER_AGENT, expected_ua)
 
+    def test_thread_local_http(self):
+        credentials = _Credentials(lambda x: object())
+        conn = self._makeOne(credentials)
+
+        self.assertTrue(conn.http is not None)
+
+        # Should return the same instance when called again.
+        self.assertTrue(conn.http is conn.http)
+
+        # Should return a different instance from a different thread.
+        http = conn.http
+
+        def test_thread():
+            self.assertTrue(conn.http is not None)
+            self.assertTrue(conn.http is not http)
+
+        thread = threading.Thread(target=test_thread)
+        thread.start()
```

Comment:
It tests only that each thread gets a unique http object. I'm not sure if there's a good way to test contention, but if you do and you think it's worthwhile then I can add it. It follows that if each thread gets a unique http instance, there should be no contention.